dropshot/src/error.rs

```diff
@@ -290,15 +304,118 @@ impl HttpError {
             error_code,
             internal_message,
             external_message,
+            headers: None,
         }
     }
 
+    /// Mutably borrow the `http::HeaderMap` associated with this error, if one
+    /// exists.
+    pub fn headers_mut(&mut self) -> Option<&mut http::HeaderMap> {
+        self.headers.as_deref_mut()
+    }
+
+    /// Borrow the `http::HeaderMap` associated with this error, if one
+    /// exists.
+    pub fn headers_ref(&self) -> Option<&http::HeaderMap> {
+        self.headers.as_deref()
+    }
```

Comment:
is this needed? I see that we don't have it for `HttpResponseHeaders`

Reply:
It's not strictly necessary, but it felt like due to the `Option<Box<...>>` situation, it was a bit nicer than making the user figure out `as_deref()` on their own. I'm fine with removing it.

Reply:
In general: I think less is more. Perhaps at least we should have these two mechanisms (i.e. HttpResponseHeaders) work similarly.

Reply:
Yeah, that's fair. I've removed `headers_ref()` in d69f141d3dbd210cfb91a92f5a8cf9a4e55562a1.

I've also changed `headers_mut()` to return `&mut HeaderMap`, creating one if it's `None`, so that it can have the same signature as `HttpResponseHeaders::headers_mut()`. I figured that most of the time, if you're looking to mutate a headermap, it's because you want there to be headers.  Users who care about the `Option` can just touch the `headers` field directly.